packages/super-components/components/msc-billing-tile/src/msc-billing-tile/msc-billing-tile.tsx

```diff
@@ -0,0 +1,279 @@
+      this.contactBilling = serviceData.contactBilling;
+      this.contactTech = serviceData.contactTech;
+      this.serviceId = serviceData.serviceId;
+      this.creationDate = formatDate(serviceData.creation, this.language);
+      if (serviceData.status === 'ok') {
+        if (serviceData.renew.deleteAtExpiration === true) {
+          // Red chip 'Cancellation requested', link in menu 'Stop cancellation of service'
+          this.renewStatus = 'deleteAtExpiration';
+        } else {
+          // service still active
+          if (
+            serviceData.renew.automatic === true &&
+            serviceData.renew.manualPayment === false
+          )
+            // Green chip 'Automatic renewal', link in menu 'Manage my commitment' and 'Cancel subscription'
+            this.renewStatus = 'automatic';
+          if (
+            serviceData.renew.automatic === false &&
+            serviceData.renew.manualPayment === true
+          )
+            // Yellow chip 'Manual renewal'
+            this.renewStatus = 'manualPayment';
+        }
+      }
+      if (serviceData.status === 'expired') {
+        // Red chip 'expired', link in menu 'Renew service'
+        this.renewStatus = 'expired';
+      }
+    });
+    await fetchServiceDetails(this.serviceId).then((serviceData) => {
+      this.nextBillingDate = formatDate(
+        serviceData.billing?.nextBillingDate,
+        this.language,
+      );
+      if (
+        serviceData.billing.engagement === null &&
+        !serviceData.billing.engagementRequest
+      ) {
+        // should be null if no commitment
+        // Red chip 'None', no link or link 'Commit'
+        this.commitmentStatus = 'none';
+      } else if (serviceData.billing.engagement?.endDate < new Date()) {
+        // Red chip 'Ended DATE', no link
+        this.commitmentStatus = 'ended';
+      } else if (serviceData.billing.engagement?.endDate > new Date()) {
+        // No chip 'Renews on DATE', link 'Re-commit and get a discount'
+        this.commitmentStatus = 'renews';
+      } else if (serviceData.billing.engagementRequest) {
+        // No chip 'Your service commitment will begin from DATE'
+        this.commitmentStatus = 'requested';
+        this.requestDate = serviceData.billing.engagementRequest.requestDate;
+      } else if (serviceData.billing.expirationDate) {
+        // No chip 'Ends DATE'
+        this.commitmentStatus = 'ends';
+      }
+    });
+    if (this.getServiceType() === 'DOMAIN')
+      await fetchDomainOwner(this.getServiceName()).then((serviceData) => {
+        this.whoisOwnerDomain = serviceData.whoisOwner;
+      });
+  }
+
+  getServiceName() {
+    const parts = this.servicePath.split('/');
+    return parts[parts.length - 1]; // get the last part
+  }
+
+  getServiceType() {
+    const parts = this.servicePath.split('/');
+    parts.pop(); // Remove the service name
+    return parts.join('_').toUpperCase(); // Join remaining parts with '_', and convert to upper case
```

Comment:
```suggestion
    return parts.join('_').toUpperCase();
```

Useless comment : we can understand this just by reading the code
We should write comments to explain **why** we are doing things, not how it is done since we can determine this by reading the code (unless the code is very complex)

Reply:
✅